src/infrastructure/blockchain/ethereum/EthereumOracleAdapter.ts

```diff
@@ -1,22 +1,36 @@
-const _ = require('lodash');
-const EventEmitter = require('events');
-const OracleGateway = require('../../../domain/blockchain/port/OracleGateway');
+import * as EventEmitter from 'events';
+import * as _ from 'lodash';
+
+import Request from '@core/domain/request/Request';
+import {Contract} from 'web3-eth-contract/types';
+import {AbiItem} from 'web3-utils/types';
+import Web3 from 'web3/types';
+import OracleGateway from '../../../domain/blockchain/port/OracleGateway';
 
 const ONE_SECOND_MILLIS = 1000;
 
-class EthereumOracleAdapter extends OracleGateway {
-  constructor(web3, abi, address) {
-    super();
+class EthereumOracleAdapter implements OracleGateway {
+  private static finishedEventName(requestId: string): string {
```

Comment:
Since this is private maybe we should move it to the bottom of the file?

Reply:
agree, private methods should be at the end